src/test/java/edu/ucsb/courses/controllers/HistorySearchControllerTests.java

```diff
@@ -0,0 +1,57 @@
+package edu.ucsb.courses.controllers;
+
+import edu.ucsb.courses.config.SecurityConfig;
+import edu.ucsb.courses.documents.Course;
+import edu.ucsb.courses.documents.CoursePage;
+import edu.ucsb.courses.repositories.ArchivedCourseRepository;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// @Import(SecurityConfig.class) applies the security rules 
+// so that /api/public/** endpoints don't require authentication.
+// Otherwise you may get authorization errors when running the test
+
```

Comment:
Oh, is this an alternative to using `@WithMockUser`? If so, I love this.

Reply:
Possibly....    depending on the circumstances.